### ISSUE TITLE
small feedback - choose component needs an expanded jar(?)

### ISSUE THREAD

**Issue by darrencruse** [jpaulm/fbp-tutorial-filter-file#6]
just another small bit of feedback

things went smooth for me till step 3 but then "Choose Component" wouldn't work

your text there kind of hinted (but if It said so explicitly I missed it?) that Choose Component needs access to the individual class files so I expanded my javafbp jar and now it's working

for me being new maybe that earlier step about the DrawFBP install needs a more explicit sister step for JavaFBP Install that tells them to expand the jar?

**Comment by jpaulm** [comment]
Some good ideas here!  Bob, could you expand the setup and/or Step03
section as Darren recommends...?

The Locate JavaFBP Jar File works for me - Bob, can you duplicate this
problem?

Thanks, and best regards,

Paul

**Comment by jpaulm** [comment]
One thought I've had: DrawFBP saves its profile stuff when it closes down -
one possible explanation for the jar file problem is if it crashed, or you
closed it down without clicking on the little 'x'.  If it didn't close down
properly, we need to know what you were doing just before!

Regards,

Paul M.

**Comment by bobcorrick** [comment]
Hi Paul,

Can I suggest that the JavaFBP Jar File location, and all other user settings that should be preserved, be saved and updated immediately to storage, to avoid loss of context when resuming after a disorderly shutdown.

I must admit I often just "close all programs" (one type of disorderly shutdown) after a session, so I've noticed this sometimes and never mentioned it.

I realise it would help you if there were some test results, to point out the areas that need to be updated.  I'll include these in the "tutorial regression test" feedback.

Before I finish the tutorial regression testing, I'll change the steps 00 and 03 as Darren suggests. Trying by iPad for the first time...

Kind regards,
Bob

On 3 Jul 2019, at 17:55, Paul Morrison <notifications@github.com<mailto:notifications@github.com>> wrote:

One thought I've had: DrawFBP saves its profile stuff when it closes down -
one possible explanation for the jar file problem is if it crashed, or you
closed it down without clicking on the little 'x'. If it didn't close down
properly, we need to know what you were doing just before!

Regards,

Paul M.

**Comment by jpaulm** [comment]
Good idea!  I was thinking I should do that!   I have made the necessary
changes, and will put it up on the Release directory later today.
Hopefully, I will have fixed Darren's other problem also!

TIA for the changes to Step00 and Step03...!

Best regards,

Paul

**Comment by bobcorrick** [comment]
Hi Paul, Darren / GitHub,

In the GitHub project 'fbp-tutorial-filter-file', the README for Step00 has been updated to clarify the Java environment, including Java Help facility and the preparation of the JavaFBP classes.

I will revisit to amend my markdown mistakes, and any outdated advice, in due course.

I'll also simplify as far as possible, depending on DrawFBP updates and releases.

By the way, I was able to reproduce the issue that Darren reported (being unable to select a JavaFBP component until the .jar file had been expanded).  So I've explained how to do that, as part of the Java setup.

________________________________

**Comment by bobcorrick** [comment]
Don't understand why you have to expand the JavaFBP jar file - works fine
for me, unexpanded!

What was the issue this addresses?  Oh, I found where Darren describes
this...

However, I have never had to do this.

Darren, could you show me where this is implied?!  What happens when you
tried to use it unexpanded?  Maybe you could raise it as an issue...?

Thanks, guys,

Paul

**Comment by jpaulm** [comment]
Sorry, Darren, for some reason I didn't understand your original post - it seems there is a bug, and what you describe is a workaround!  I never ran across it in testing as I had added the JavaFBP jar file to my PATH variable a long time ago, and never moved it, so my stuff worked!  Apologies to you and Bob!

Looking back, I see you said "wouldn't work" - in future maybe you could be more specific...!

**Comment by jpaulm** [comment]
Darren, me again!  I just spent 1/2 a day apparently barking up the wrong tree!  Could you let us know what was the bug you ran into?

Thx,

Paul

**Comment by darrencruse** [comment]
Well gosh sorry about this I just removed my expanded jar and tried Choose Component/Subnet Class against the unexpanded jar and it did work for me.

All I can figure is the "Files of Type" dropdown was initially on "Class files" and I changed it just now before I was able to choose the jar file.

I guess the other day I didn't notice/think/understand to change that Files of Type dropdown and jumped to the conclusion I needed to expand the jar(!!).
